base/complex.jl

```diff
@@ -996,7 +996,8 @@ function round(z::Complex, rr::RoundingMode=RoundNearest, ri::RoundingMode=rr; k
     Complex(round(real(z), rr; kwargs...),
             round(imag(z), ri; kwargs...))
 end
-
+round(::Type{Complex{T}}, x::Complex{U}) where U<:Real where T<:Real = Complex{T}(round(T, x.re), round(T, x.im))
+round(::Type{Complex{T}}, x::U) where U<:Real where T<:Real = Complex{T}(round(x))
```

Comment:
Since you're only using `T` here, you can make this
```julia
round(::Type{Complex{T}}, x::Complex) where {T<:Real} = Complex{T}(round(T, x.re), round(T, x.im))
round(::Type{Complex{T}}, x::Real) where {T<:Real} = Complex{T}(round(x))
```